### ISSUE TITLE
No reads_per_umi_per_cell in cell.counts.rds after dropEst

### ISSUE THREAD

**Issue by zh542370159** [hms-dbmi/dropEst#33]
I have successfully run dropEst (v0.8.3) on my .bam file generated by 10X cellranger (2.1.1)
the command is 
`dropest -f -w -u -V -G 20 -c /home/reprod/bin/dropEst/configs/10x.xml ./possorted_genome_bam.bam`
Log file looks good with no serious problem.

```
dropest -f -w -u -V -G 20 -c /home/reprod/bin/dropEst/configs/10x.xml ./possorted_genome_bam.bam 
Version: 0.8.3.
Run: 07/18/2018 21:19:30.
No merge selected.

Start parsing of bams: 21:19:30.
ERROR: can't find chromosome, id: -1
./possorted_genome_bam.bam: 4673842 total reads; 26.6% intergenic; 68.2% touch exon; 0% touch intron; 0% touch not annotated regions; can't parse 0.0305316% reads; 348073 CBs read
Bams parsed: 21:20:23.

13075 CBs with more than 20 genes
top CBs:
1531	CAGCGACCATGTCCTC
1504	ACAGCTAGTAAATGTG
1494	GACGTTAAGCCAGGAT
1448	GTTCGGGAGGGTTTCT
1422	CACACCTCAGCATGAG
1417	GAGGTGAGTTTGGGCC
1376	ACACCAATCTCTTATG
1357	CTAATGGAGAAGGACA
1353	TCAGGTATCTGTCTCG

Start merge: 21:20:23.
Merge initialized: 21:20:23.
Merge finished: 21:20:23.
Start UMI merge: 21:20:23.
13075 cells processed. Merged 58 UMIs from 58 cells.
UMI merge finished: 21:20:24.
13075 cells are considered as real.

13075 CBs with more than 20 genes, which have UMIs of the requested type.
top CBs:
1531	CAGCGACCATGTCCTC
1504	ACAGCTAGTAAATGTG
1494	GACGTTAAGCCAGGAT
1448	GTTCGGGAGGGTTTCT
1422	CACACCTCAGCATGAG
1417	GAGGTGAGTTTGGGCC
1376	ACACCAATCTCTTATG
1357	CTAATGGAGAAGGACA
1353	TCAGGTATCTGTCTCG

Done: 21:20:25.

20783 genes
top genes:
ENSG00000122304	42526
ENSG00000175646	21533
ENSG00000065978	9621
ENSG00000106013	7076
ENSG00000167468	7074
ENSG00000010318	6500
ENSG00000163467	6138
ENSG00000167526	6100
ENSG00000170315	5772
ENSG00000198804	5054

Compiling diagnostic stats: 
Reads per chromosome per cell;
Fill exon results
Fill intron results
Fill intergenic results
Saturation info;
Mean reads per UMI;
Merge targets;
Completed.

Compiling raw count matrix: 21:20:29.
20783 genes, 13075 cells.
Done: 21:20:29.
Compiling filtered count matrix: 21:20:29.
20783 genes, 13075 cells.
Done: 21:20:30.

Writing R data to cell.counts.rds ...: 21:20:30.
Completed: 21:20:39.
Writing cell.counts.mtx ...
Completed.

Compiling intron/exon matrices: 21:20:40.
Exon
20783 genes, 13075 cells.
Intron
0 genes, 13075 cells.
Intron/exon spanning
0 genes, 13075 cells.
Done: 21:20:42.

Writing R data to cell.counts.matrices.rds ...: 21:20:42.
Completed: 21:20:43.
All done: 21:20:45.
```

 However, command `dropReport.Rsc ./cell.counts.rds` ran with error:
```
Quitting from lines 118-122 (report.Rmd)
Error: 'ds.mincount' is not an exported object from 'namespace:preseqR'
In addition: Warning messages:
1: In bind_rows_(x, .id) : Unequal factor levels: coercing to character
2: In bind_rows_(x, .id) :
  binding character and factor vector, coercing into character vector
3: In bind_rows_(x, .id) :
  binding character and factor vector, coercing into character vector
4: In bind_rows_(x, .id) : Unequal factor levels: coercing to character
5: In bind_rows_(x, .id) :
  binding character and factor vector, coercing into character vector
6: In bind_rows_(x, .id) :
  binding character and factor vector, coercing into character vector
7: In GetUmisDistributionOld(reads.per.umi.per.cb, smooth = smooth) :
  UMIs distribution is empty
8: Removed 2 rows containing missing values (geom_bar).
9: In PredictKDE(clf, bc.df.pca, bandwidth.mult = kde.bandwidth.mult) :
  Found 2 points with 0 density. Either it's outliers or you should increase bandwidth. Setting score for these points to 0.5.

Execution halted
```
Some results have been generated. One of them appears wrong.
![image](https://user-images.githubusercontent.com/24678781/42887196-f52d0f2c-8ad7-11e8-8098-ef93abe1b4e7.png)

So I guess it may be caused by the cell.counts.rds. It does not have **reads_per_umi_per_cell**.
This problem appeared in the older version  #29 . And I tried to re-run dropEst but the cell.counts.rds but it  had no change 
Here is the fields in cell.counts.rds
```
d <- readRDS("./cell.counts.rds")
names(d)
 [1] "cm"                      "cm_raw"                  "reads_per_chr_per_cells" "mean_reads_per_umi"     
 [5] "saturation_info"         "merge_targets"           "aligned_reads_per_cell"  "aligned_umis_per_cell"  
 [9] "requested_umis_per_cb"   "requested_reads_per_cb" 
```

**Comment by zh542370159** [comment]
It seems that a module does not run in dropest.
If it runs successfully, the log file should seem like this:
```
Compiling diagnostic stats: 
Reads per chromosome per cell;
Fill exon results
Fill intron results
Fill intergenic results
Saturation info;
Mean reads per UMI;
**Reads per UMI per gene;**
Merge targets;
Completed.
```
So my log file does not have line "Reads per UMI per gene;"
Can give me some advice?

**Comment by zh542370159** [comment]
When I run dropest without **"-u"** , cell.counts.rds has the field **reads_per_umi_per_cell**
```
names(d)
 [1] "cm"                      "cm_raw"                  "reads_per_chr_per_cells" "mean_reads_per_umi"     
 [5] "saturation_info"         "merge_targets"           "aligned_reads_per_cell"  "aligned_umis_per_cell"  
 [9] "requested_umis_per_cb"   "requested_reads_per_cb"  "reads_per_umi_per_cell" 
```
However,  the error still occurs...
```
label: unnamed-chunk-12 (with options)
List of 2
 $ fig.width : num 4.5
 $ fig.height: num 4

  |..............................................                   |  71%
  ordinary text without R code

  |................................................                 |  74%
label: unnamed-chunk-13 (with options)
List of 4
 $ message   : logi TRUE
 $ warning   : logi FALSE
 $ fig.width : num 4.5
 $ fig.height: num 4

Quitting from lines 118-122 (report.Rmd)
Error: 'ds.mincount' is not an exported object from 'namespace:preseqR'
In addition: Warning messages:
1: In bind_rows_(x, .id) : Unequal factor levels: coercing to character
2: In bind_rows_(x, .id) :
  binding character and factor vector, coercing into character vector
3: In bind_rows_(x, .id) :
  binding character and factor vector, coercing into character vector
4: In bind_rows_(x, .id) : Unequal factor levels: coercing to character
5: In bind_rows_(x, .id) :
  binding character and factor vector, coercing into character vector
6: In bind_rows_(x, .id) :
  binding character and factor vector, coercing into character vector
7: Removed 2 rows containing missing values (geom_bar).
8: In PredictKDE(clf, bc.df.pca, bandwidth.mult = kde.bandwidth.mult) :
  Found 2 points with 0 density. Either it's outliers or you should increase bandwidth. Setting score for these points to 0.5.

Execution halted
```

**Comment by zh542370159** [comment]
I think I found the problem.
In function **dropestr::EstimateSaturation()**, **ds.mincount()** in preseqR package is required.
However, **preseqR (latest v4.0.0)** has no function **ds.mincount()** instead of **ds.rSAC()**.
So I modified the **quality_control.R** script and then it works fine.
The following is the part of the script modified:
```
EstimateSaturation <- function(reads.by.umig.vec, reads.by.umig.cbs, umi.counts, steps.num=100, max.estimate.rate=10, top.cells=1000) {
  if (!requireNamespace("preseqR", quietly = TRUE))
    stop("preseqR package is required")

  top.cbs <- names(umi.counts)[1:top.cells]
  top.reads.by.umig <- reads.by.umig.vec[reads.by.umig.cbs %in% top.cbs]
  counts <- ValueCounts(top.reads.by.umig)
  counts <- counts[order(as.integer(names(counts)))]
  freqs <- as.integer(names(counts))

  estimator <- preseqR::ds.rSAC(cbind(freqs, counts))
  size.ratios <- seq(0, max.estimate.rate, length.out=steps.num)
  sequencing.depth <- sum(top.reads.by.umig) * size.ratios

  estimates <- sapply(size.ratios, estimator)
  return(list(sat=data.frame(estimates, depth=sequencing.depth),
              current=data.frame(depth=sum(top.reads.by.umig), estimates=length(top.reads.by.umig))))
}
```

**Comment by VPetukhov** [comment]
Hi! Thank you for such detailed analysis of the problem!

> So my log file does not have line "Reads per UMI per gene;"
> Can give me some advice?

`-u` option prevents output of `reads_per_umi_per_cell`, it was done purposely. I just added a note about this in documentation. Though I still need to fix the report according to it.

>  However, preseqR (latest v4.0.0) has no function ds.mincount() instead of ds.rSAC().

Looks like preseqR dropped backward compatibility again. Thank you for the fix! I'll test it and update the code.

**Comment by VPetukhov** [comment]
Added your fix in 9a2af41acb9db77bafe07fcaf9b35583adfd1c0e. Thanks!
